Makefile

```
@@ -169,9 +169,11 @@ lint-fix: golangci-lint
 shell-lint: ## Run shell linting.
 	$(PROJECT_DIR)/hack/shellcheck/verify.sh
 
+PATHS_TO_VERIFY := config/components apis charts/kueue/templates client-go site/
 .PHONY: verify
 verify: gomod-verify ci-lint fmt-verify shell-lint toc-verify manifests generate update-helm generate-apiref generate-kueuectl-docs prepare-release-branch
-	git --no-pager diff --exit-code config/components apis charts/kueue/templates client-go site/
+	git --no-pager diff --exit-code $(PATHS_TO_VERIFY)
+	if git ls-files --exclude-standard --others $(PATHS_TO_VERIFY) | grep -q . ; then exit 1; fi
```

```suggestion
	git --no-pager diff --exit-code $(PATHS_TO_VERIFY)
	git --cashed --no-pager diff --exit-code $(PATHS_TO_VERIFY)
```

Instead of `ls-files` command, couldn't we check just staged files like this, right?

---

`--cached` flag doesn't allow to check not added to git files.

---

That's right. So, I added 2 commands, `git diif` and `git diff --cached` as you can see above.